samples/net/sockets/echo_async_select/src/socket_echo_select.c

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <errno.h>
+#include <stdlib.h>
+
+#ifndef __ZEPHYR__
+
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+#else
+
+#include <fcntl.h>
+#include <net/socket.h>
+#include <kernel.h>
+#include <net/net_app.h>
+
+#endif
+
+/* For Zephyr, keep max number of fd's in sync with max poll() capacity */
+#ifdef CONFIG_NET_SOCKETS_POLL_MAX
+#define NUM_FDS CONFIG_NET_SOCKETS_POLL_MAX
+#else
+#define NUM_FDS 5
+#endif
+
+#define PORT 4242
+
+/* Number of simultaneous client connections will be NUM_FDS be minus 2 */
+fd_set readfds;
+fd_set workfds;
+int fdnum;
+
+#define fatal(msg, ...) { \
+		printf("Error: " msg "\n", ##__VA_ARGS__); \
+		exit(1); \
+	}
+
+
+static void setblocking(int fd, bool val)
+{
+	int fl, res;
+
+	fl = fcntl(fd, F_GETFL, 0);
+	if (fl == -1) {
+		fatal("fcntl(F_GETFL): %d", errno);
+	}
+
+	if (val) {
+		fl &= ~O_NONBLOCK;
+	} else {
+		fl |= O_NONBLOCK;
+	}
+
+	res = fcntl(fd, F_SETFL, fl);
+	if (fl == -1) {
+		fatal("fcntl(F_SETFL): %d", errno);
+	}
+}
+
+int pollfds_add(int fd)
+{
+	FD_SET(fd, &readfds);
+
+	if (fd >= fdnum) {
+		fdnum = fd + 1;
+	}
+
+	return 0;
+}
+
+void pollfds_del(int fd)
+{
+	FD_CLR(fd, &readfds);
+}
+
+int main(void)
+{
+	int res;
+	static int counter;
+	int serv4, serv6;
+	struct sockaddr_in bind_addr4 = {
+		.sin_family = AF_INET,
+		.sin_port = htons(PORT),
+		.sin_addr = {
+			.s_addr = htonl(INADDR_ANY),
+		},
+	};
+	struct sockaddr_in6 bind_addr6 = {
+		.sin6_family = AF_INET6,
+		.sin6_port = htons(PORT),
+		.sin6_addr = IN6ADDR_ANY_INIT,
+	};
+
+	serv4 = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+	res = bind(serv4, (struct sockaddr *)&bind_addr4, sizeof(bind_addr4));
+	if (res == -1) {
+		printf("Cannot bind IPv4, errno: %d\n", errno);
+	}
+
+	serv6 = socket(AF_INET6, SOCK_STREAM, IPPROTO_TCP);
+	#ifdef IPV6_V6ONLY
+	/* For Linux, we need to make socket IPv6-only to bind it to the
+	 * same port as IPv4 socket above.
+	 */
+	int TRUE = 1;
```

Comment:
This looks very confusing, just use normal variable name. Typically in many linux apps, the variable is named "on" when setting a true value.

Reply:
> just use normal variable name

Again, just copy of socket_echo.c